schemas/README.md

```diff
@@ -1593,6 +2172,548 @@ Raw image data
 </tr>
 </table>
 
+## SceneEntity
+
+A visual element in a 3D scene. An entity may be composed of multiple primitives which all share the same frame of reference.
+
+<table>
+  <tr>
+    <th>field</th>
+    <th>type</th>
+    <th>description</th>
+  </tr>
+<tr>
+<td><code>timestamp</code></td>
+<td>
+
+time
+
+</td>
+<td>
+
+Timestamp of the entity
+
+</td>
+</tr>
+<tr>
+<td><code>frame_id</code></td>
+<td>
+
+string
+
+</td>
+<td>
+
+Frame of reference
+
+</td>
+</tr>
+<tr>
+<td><code>id</code></td>
+<td>
+
+string
+
+</td>
+<td>
+
+Identifier for the entity. A entity will replace any prior entity on the same topic with the same `id`.
+
+</td>
+</tr>
+<tr>
+<td><code>lifetime</code></td>
+<td>
+
+duration
+
+</td>
+<td>
+
+Length of time (relative to `timestamp`) after which the entity should be automatically removed. Zero value indicates the entity should remain visible until it is replaced or deleted.
+
+</td>
+</tr>
+<tr>
+<td><code>frame_locked</code></td>
+<td>
+
+boolean
+
+</td>
+<td>
+
+Whether the entity should keep its location in the fixed frame (false) or follow the frame specified in `frame_id` as it moves relative to the fixed frame (true)
+
+</td>
+</tr>
+<tr>
+<td><code>metadata</code></td>
+<td>
+
+[KeyValuePair](#keyvaluepair)[]
+
+</td>
+<td>
+
+Additional user-provided metadata associated with the entity. Keys must be unique.
+
+</td>
+</tr>
+<tr>
+<td><code>arrows</code></td>
+<td>
+
+[ArrowPrimitive](#arrowprimitive)[]
+
+</td>
+<td>
+
+Arrow primitives
+
+</td>
+</tr>
+<tr>
+<td><code>cubes</code></td>
+<td>
+
+[CubePrimitive](#cubeprimitive)[]
+
+</td>
+<td>
+
+Cube primitives
+
+</td>
+</tr>
+<tr>
+<td><code>spheres</code></td>
+<td>
+
+[SpherePrimitive](#sphereprimitive)[]
+
+</td>
+<td>
+
+Sphere primitives
+
+</td>
+</tr>
+<tr>
+<td><code>cylinders</code></td>
+<td>
+
+[CylinderPrimitive](#cylinderprimitive)[]
+
+</td>
+<td>
+
+Cylinder primitives
+
+</td>
+</tr>
+<tr>
+<td><code>lines</code></td>
+<td>
+
+[LinePrimitive](#lineprimitive)[]
+
+</td>
+<td>
+
+Line primitives
+
+</td>
+</tr>
+<tr>
+<td><code>triangles</code></td>
+<td>
+
+[TriangleListPrimitive](#trianglelistprimitive)[]
+
+</td>
+<td>
+
+Triangle list primitives
+
+</td>
+</tr>
+<tr>
+<td><code>texts</code></td>
+<td>
+
+[TextPrimitive](#textprimitive)[]
+
+</td>
+<td>
+
+Text primitives
+
+</td>
+</tr>
+<tr>
+<td><code>models</code></td>
+<td>
+
+[ModelPrimitive](#modelprimitive)[]
+
+</td>
+<td>
+
+Model primitives
+
+</td>
+</tr>
+</table>
+
+## SceneEntityDeletion
+
+Command to remove previously published entities
+
+<table>
+  <tr>
+    <th>field</th>
+    <th>type</th>
+    <th>description</th>
+  </tr>
+<tr>
+<td><code>timestamp</code></td>
+<td>
+
+time
+
+</td>
+<td>
+
+Timestamp of the deletion. Only matching entities earlier than this timestamp will be deleted.
+
+</td>
+</tr>
+<tr>
+<td><code>type</code></td>
+<td>
+
+[enum SceneEntityDeletionType](#enum-sceneentitydeletiontype)
```

Comment:
This could be a simple `delete_all` bool. Doesn't save many bytes, but it seems simpler to me when reading the docs, rather than clicking through into a whole enum with only two values.

Reply:
This was leaving it open for a third option to delete all entities under the same `namespace`, so depends how confident we are about the omission of the `namespace` feature. (Also I don't think we should design our API just around the simplicity of the docs, since there are also things we could do to improve the docs experience here, such as embedding the enum values directly in the message field docs.)

Reply:
👍 don't feel strongly about this